### PR TITLE
Grid bulk delete confirmation modal - Monitoring

### LINKED ISSUE #17845: Add a modal to confirm or cancel delete in bulk actions [2/2]
Add a modal to confirm or cancel delete in bulk action instead of the browser one

Already migrated pages for which it must be added:
- [x] Catalog > Monitoring
- [x] Advanced parameters > Email  
- [x] Advanced parameters > Team > Employees  
- [x] Advanced parameters > Team > Profiles  
- [x] Advanced parameters > Database > SQL Manager  
- [x] Advanced parameters > Database > DB Backup
- [x] Advanced parameters > Webservice 
- [x] Shop parameters > Contact > Contacts
- [x] Shop parameters > Traffic & SEO > SEO&URLs

### PR DESCRIPTION
**Pull request by sowbiba**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add a confirmation modal when deleting multiple rows from a products monitoring grid
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Partially fixes #17845 
| How to test?  | Go to Catalog > Monitoring in the BO, Select multiple rows, try to delete a row, you will have a bootstrap modal to confirm deletion

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21050)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/pages/monitoring/index.js [hunk 1]
```diff
@@ -40,6 +41,8 @@ import ExportToSqlManagerExtension
   from '@components/grid/extension/export-to-sql-manager-extension';
 import ShowcaseCard from '@components/showcase-card/showcase-card';
 import ShowcaseCardCloseExtension from '@components/showcase-card/extension/showcase-card-close-extension';
+import BulkActionCheckboxExtension from '@components/grid/extension/bulk-action-checkbox-extension';
+
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/pages/monitoring/index.js:45]:
```suggestion
```

File: src/Adapter/Product/CommandHandler/BulkDeleteProductHandler.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
```
Comment (Progi1984) [inline src/Adapter/Product/CommandHandler/BulkDeleteProductHandler.php]:
```suggestion
 */

declare(strict_types=1);

namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
```

File: src/Adapter/Product/CommandHandler/BulkDeleteProductHandler.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws BulkDeleteProductException
+     */
+    public function handle(BulkDeleteProductCommand $command)
+    {
+        $productIds = array_map(function (ProductId $productId) { return $productId->getValue(); }, $command->getProductIds());
```
Comment (Progi1984) [inline src/Adapter/Product/CommandHandler/BulkDeleteProductHandler.php]:
```suggestion
        $productIds = array_map(function (ProductId $productId) {
            return $productId->getValue();
        }, $command->getProductIds());
```

File: src/Core/Domain/Product/Command/BulkDeleteProductCommand.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Command;
```
Comment (Progi1984) [inline src/Core/Domain/Product/Command/BulkDeleteProductCommand.php]:
```suggestion
 */

declare(strict_types=1);

namespace PrestaShop\PrestaShop\Core\Domain\Product\Command;
```

File: src/Core/Domain/Product/Command/BulkDeleteProductCommand.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+     * @var ProductId[]
+     */
+    private $productIds;
+
+    /**
+     * @param int[] $productIds
+     *
+     * @throws ProductConstraintException
+     */
+    public function __construct($productIds)
```
Comment (Progi1984) [inline src/Core/Domain/Product/Command/BulkDeleteProductCommand.php]:
```suggestion
    public function __construct(array $productIds)
```

File: src/Core/Domain/Product/Command/BulkDeleteProductCommand.php [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        $this->setProductIds($productIds);
+    }
+
+    /**
+     * @return ProductId[]
+     */
+    public function getProductIds()
```
Comment (Progi1984) [inline src/Core/Domain/Product/Command/BulkDeleteProductCommand.php]:
```suggestion
    public function getProductIds(): array
```

File: src/Core/Domain/Product/Exception/BulkDeleteProductException.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Exception;
```
Comment (Progi1984) [inline src/Core/Domain/Product/Exception/BulkDeleteProductException.php]:
```suggestion
 */

declare(strict_types=1);

namespace PrestaShop\PrestaShop\Core\Domain\Product\Exception;
```

File: src/Adapter/Product/CommandHandler/BulkDeleteProductHandler.php [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+ * Handles command which deletes addresses in bulk action
+ */
+final class BulkDeleteProductHandler implements BulkDeleteProductHandlerInterface
+{
+    /**
+     * @var AdminProductDataUpdater
+     */
+    private $adminProductDataUpdater;
+
+    public function __construct(AdminProductDataUpdater $adminProductDataUpdater)
```
Comment (Progi1984) [inline src/Adapter/Product/CommandHandler/BulkDeleteProductHandler.php]: PHPDoc ;)

**Review by Progi1984 — COMMENTED** [review]
A last comment ;)

**Review by atomiix — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by sowbiba — COMMENTED** [review]
Bulk delete on disabled products fixed @SD1982

**Review by atomiix — DISMISSED** [review]

**Comment by SD1982** [comment]
Hi @sowbiba Can you please resolve the conflicting files ?

File: src/Core/Domain/Product/Exception/BulkDeleteProductException.php [hunk 9]
```diff
@@ -0,0 +1,63 @@
+<?php
```
Comment (atomiix) [inline src/Core/Domain/Product/Exception/BulkDeleteProductException.php]: I don't see where this class is used 🤔

Comment (Progi1984) [reply src/Core/Domain/Product/Exception/BulkDeleteProductException.php]: @sowbiba Needs an answer

Comment (sowbiba) [reply src/Core/Domain/Product/Exception/BulkDeleteProductException.php]: It's not used and I cant remember why I added this class, so it's removed

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM Thanks @sowbiba !!

**Comment by Progi1984** [comment]
Thanks @sowbiba & @SD1982

**Pull request merged**